doc/devElements.md

```diff
@@ -100,7 +100,7 @@ Function | Return object | modifiable `data` keys | unmodifiable `data` keys | D
 --- | --- | --- | --- | ---
 `generate()` | `data` (dict) | `params`, `correct_answers` | `variant_seed`, `options` | Generate the parameter and true answers for a new random question variant. Set `data["params"][name]` and `data["correct_answers"][name]` for any variables as needed. Return the modified `data` dictionary.
 `prepare()` | `data` (dict) | `params`, `correct_answers` | `variant_seed`, `options` | Final question preparation after element code has run. Can modify data as necessary. Return the modified `data` dictionary.
-`render()` | `html` (string) | `params`, `correct_answers`, `submitted_answers`, `format_errors`, `partial_scores`, `score`, `feedback` `variant_seed`, `options`, `raw_submitted_answers`, `editable`, `panel` | Render the HTML for one panel and return it as a string.
+`render()` | `html` (string) | | `params`, `correct_answers`, `submitted_answers`, `format_errors`, `partial_scores`, `score`, `feedback` `variant_seed`, `options`, `raw_submitted_answers`, `editable`, `panel` | Render the HTML for one panel and return it as a string.
```

Comment:
Column was missing. I believe I added the missing column back to the right place -- based on my interpretation of [freeform.js#L364](https://github.com/PrairieLearn/PrairieLearn/blob/master/question-servers/freeform.js#L364), all keys are unmodifiable in `render()`.

Reply:
That's correct, they are all unmodifiable. The code is `freeform.js` is the ultimate authority on this, because it actually enforces these restrictions 😄